pkg/catalog/catalog.go

```diff
@@ -102,7 +105,18 @@ func getRegistryDevfiles(registry Registry) ([]DevfileComponentType, error) {
 	}
 	registry.URL = URL
 	indexLink := registry.URL + indexPath
-	jsonBytes, err := util.HTTPGetRequest(indexLink)
+	token := ""
+	if registryUtil.IsSecure(registry.Name) {
+		token, err = keyring.Get(util.CredentialPrefix+registry.Name, "default")
+		if err != nil {
+			return nil, errors.Wrap(err, "Unable to get secure registry credential from keyring")
+		}
+	}
+	request := util.HTTPRequestParams{
```

Comment:
Nit: It might be cleaner to declare the params struct above the IsSecure check (without the token field) and then assign the value of `token` inside the if-statement. So something like:
```
token, err = keyring.Get(util.CredentialPrefix+registry.Name, "default")
if err != nil {
    return nil, errors.Wrap(err, "Unable to get secure registry credential from keyring")
}
request.Token = token
```

Reply:
Good suggestion, I will refactor it.